derp/derp_server.go

```diff
@@ -141,6 +142,11 @@ type Server struct {
 	// because it includes intra-region forwarded packets as the
 	// src.
 	sentTo map[key.Public]map[key.Public]int64 // src => dst => dst's latest sclient.connNum
+
+	addrOfKey map[netaddr.IPPort]key.Public
+
+	statMu       sync.Mutex
```

Comment:
in general, only have one mutex on a type. otherwise you need to aggressively document & follow lock ordering rules if any caller code ever needs to update/read multiple things under different mutex. Otherwise you can deadlock.